warehouse/query-core/src/main/java/datawave/query/function/JexlContextCreator.java

```diff
@@ -64,7 +65,47 @@ public JexlContextCreator() {
             log.trace("Constructed context from index and attribute Documents: " + context);
         }
         
-        return Tuples.tuple(from.first(), from.second(), context);
+        return Tuples.tuple(dedupedFrom.first(), dedupedFrom.second(), context);
+    }
+    
+    private Tuple3 dedupeAttributes(Tuple3<Key,Document,Map<String,Object>> rawTuple) {
+        Map<String,Attribute<? extends Comparable<?>>> dict = rawTuple.second().getDictionary();
+        Set<Attribute<? extends Comparable<?>>> dedupedAttributeSet = new HashSet<Attribute<? extends Comparable<?>>>();
+        
+        for (Entry<String,Attribute<? extends Comparable<?>>> entry : dict.entrySet()) {
+            if (entry.getKey().equals(BODY_KEY)) {
```

Comment:
Why are we only doing this for the BODY_KEY?  The BODY values are in fact index only so there will be no values from the event.  Dedupping those values are reasonable but why not do that for all fields?

Reply:
Updated to check all keys